app/services/proxy_api/resource.rb

```diff
@@ -67,7 +67,6 @@ def parse(response)
     # Perform GET operation on the supplied path
     def get(path = nil, payload = {})
       query = payload.delete(:query)
-      Foreman::Deprecation.deprecation_warning("3.3", "passing additional headers to ProxyApi resource GET action") unless payload.empty?
```

Comment:
I think you should remove the code that deals with a non-empty payload.

Reply:
hmm, then to get rid of the payload for get method it looks some of the classes inherited from `ProxyAPI::Resource` needs change. I will git it a try